dulwich/tests/test_repository.py

```diff
@@ -765,3 +766,15 @@ def test_commit_no_encode_decode(self):
             mode, id = tree_lookup_path(r.get_object, r[commit_sha].tree, name)
             self.assertEqual(stat.S_IFREG | 0o644, mode)
             self.assertEqual(encoding.encode('ascii'), r[id].data)
+
+    def test_discover_intended(self):
+        path = os.path.join(self._repo_dir, 'b/c')
+        r = Repo.discover(path)
+        self.assertEqual(r.head(), self._repo.head())
+
+    def test_discover_isrepo(self):
+        r = Repo.discover(self._repo_dir)
+        self.assertEqual(r.head(), self._repo.head())
+
+    def test_discover_notrepo(self):
+        self.assertRaises(NotGitRepository, Repo.discover('/'))
```

Comment:
This test errors with:
```
ERROR: test_discover_notrepo (dulwich.tests.test_repository.BuildRepoRootTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "dulwich/tests/test_repository.py", line 780, in test_discover_notrepo
    self.assertRaises(NotGitRepository, Repo.discover('/'))
  File "dulwich/repo.py", line 697, in discover
    "No git repository was found at %(path)s" % dict(path=start)
dulwich.errors.NotGitRepository: No git repository was found at /
----------------------------------------------------------------------
```
Is there something wrong with my assertion? I generally use py.test `:)`